ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -407,9 +409,38 @@ boolean isStarted() {
     return started;
   }
 
+  /**
+   * Compacts the store data based on {@code details}.
+   * @param details the {@link CompactionDetails} describing what needs to be compacted.
+   * @throws IllegalArgumentException if any of the provided segments doesn't exist in the log or if one or more offsets
+   * in the segments to compact are in the journal.
+   * @throws IOException if there is any error creating the {@link CompactionLog}.
+   * @throws StoreException if there are any errors during the compaction.
+   */
+  void compact(CompactionDetails details) throws IOException, StoreException {
+    checkStarted();
+    // TODO: compactor.compact(details);
```

Comment:
can we log statements for now. I know you are thinking about more logging and metrics in a new patch. Just incase if you think in the mean time, very minimal log statements (resumeCompaction and compact calls)